libcron/include/libcron/Task.h

```diff
@@ -1,26 +1,56 @@
 #pragma once
 
 #include <functional>
-#include "CronData.h"
-#include "CronSchedule.h"
 #include <chrono>
 #include <utility>
+#include "CronData.h"
+#include "CronSchedule.h"
 
 namespace libcron
 {
-    class Task
+    class TaskInformation
+    {
+        public:
+            virtual ~TaskInformation() {}
+            virtual std::chrono::system_clock::duration get_delay() const = 0;
+    };
+
+    class Task : public TaskInformation
     {
         public:
+            using TaskFunction = std::function<void(const TaskInformation*)>;
 
-            Task(std::string name, const CronSchedule schedule, std::function<void()> task)
+            class TaskProxy
+            {
+                public:
+                TaskProxy(TaskFunction task) : task(std::move(task)) {}
+
+                void operator() (const TaskInformation* i)
+                {
+                    task(i);
+                }
+
+                private:
+                TaskFunction task;
+            };
+
+            Task(std::string name, const CronSchedule schedule, TaskProxy task)
                     : name(std::move(name)), schedule(std::move(schedule)), task(std::move(task))
             {
             }
 
             void execute(std::chrono::system_clock::time_point now)
             {
+                // Next Schedule is still the current schedule, check if execution was on time (within 1 second) 
```

Comment:
The delay is actually more accurate than full seconds.

Reply:
Done (improved comment)